Review branch1

ahngh0113 opened the pull request.

ahngh0113 committed:
feat: create index.html and add header tag, container-left

change lang, en to ko
add semantic tag(header)
add navbar tag
add menu names
add image, github logo

ahngh0113 committed:
feat: link reset.min.css

add link reset.min.css that head tag

ahngh0113 committed:
feat: connect css file

connect style.css

ahngh0113 committed:
style: create style.css file and apply header and container-left  of index.html

add function that show submenu when mouse hover
add function that change mouse icon when mouse hover
change html structure

TODO

apply container-right css

ahngh0113 committed:
feat: add container-right html

add input tag
add signup and signin button

ahngh0113 committed:
style: apply container-right css

change background when input tag focue
chnage button color when mouse hover
add button border color

ahngh0113 committed:
docs: add images

add logo image
add github_header image

ahngh0113 committed:
docs: add image

add github header image

ahngh0113 committed:
docs: change image address

change github header image address

ahngh0113 committed:
docs: add image

add github header image I created

ahngh0113 committed:
docs: add image and change the structure of a paragraph

add github header image I created
add 'TODO' paragraph

heejinlee07 commented on index.html:
```diff
@@ -0,0 +1,193 @@
+<!DOCTYPE html>
+<html lang="ko">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/reset-css@5.0.1/reset.min.css"
+    />
+    <link rel="stylesheet" href="./style.css" />
+    <title>github header</title>
+  </head>
```
`<Head>`영역에 favicon을 추가하면 좋습니다. 
일반적으로 모든 사이트에서는 favicon이 사용되고 있습니다. 
자세한 사용방법은 아래 링크의 글을 읽어보시기 바랍니다. 
favicon외에도 `<Head>`영역에 필수로 사용되는 태그들에 대한 정보가 있습니다.

[https://developer.mozilla.org/ko/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML#맞춤_아이콘_추가하기](https://developer.mozilla.org/ko/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML#%EB%A7%9E%EC%B6%A4_%EC%95%84%EC%9D%B4%EC%BD%98_%EC%B6%94%EA%B0%80%ED%95%98%EA%B8%B0)

heejinlee07 replied:
알고 작성하셨을 수도 있는데 `reset.css`를 먼저 작성하고, 그 다음으로 `style.css`를 작성하신 점이 좋습니다. 항상 `reset.css`가 다른 스타일파일보다 먼저 호출되어야 작성된 내용을 덮어쓰지 않습니다.

heejinlee07 commented on index.html:
```diff
@@ -0,0 +1,193 @@
+                <input type="text" class="input" placeholder="Search Github" />
+              </div>
+            </li>
+            <li class="menus">
+              <div class="menus__name">Sign in</div>
+            </li>
+            <li class="menus signup">
+              <div class="menus__name">Sign up</div>
```
전반적으로 시멘틱하게 작성해주셨는데, Sign in, Sign up도 시멘틱을 고려하여 `<button>`으로 작성해보시는 것도 좋습니다.

heejinlee07 commented on style.css:
```diff
@@ -0,0 +1,213 @@
+/* header tag*/
+
+.header {
+  width: 100px;
+  background-color: #242a2e;
+  color: #fff;
+  display: flex;
+  width: 100%;
+  height: 72px;
+  justify-content: center;
+  align-items: center;
```
속성을 작성할 때 코딩컨벤션을 고려햐여 작성하시는 것이 좋습니다. 이 경우 코드 가장 상단에 있어야 하는 속성은 `display: flex`입니다. css 속성 기술시 컨벤션을 따르지 않는다고하여 사이트 이펙트가 발생하는 것은 아니지만, 실무에서도 되도록 컨벤션을 따라서 작성하도록 되어있고, 컨벤션을 따르는 것이 보다 가독성 좋은 코드를 작성하게 합니다. 관련된 내용은 아래를 참고하세요:)

https://www.beautifulcss.com/archives/203

heejinlee07 commented on style.css:
```diff
@@ -0,0 +1,213 @@
+.header .container .container-right .nav .inputdiv .input:focus {
+  background-color: white;
+  border-radius: 0;
+  border-radius: 5px 5px 0px 0px;
```
`border-radius`도 `border-radius: 5px  0px;`으로 축약하여 작성 가능합니다.

heejinlee07 commented on style.css:
```diff
@@ -0,0 +1,213 @@
+  cursor: pointer;
+}
+
+.header .container .container-left .nav .menus .menus__name {
+  font-weight: 600;
+}
+
+.header .container .container-left .nav .menus .subMenu {
+  display: none;
+}
+
+.header .container .container-left .nav .menus .subMenu .list {
+  width: 100%;
+}
+
+.header .container .container-left .nav .menus .subMenu .list .menu {
+  padding: 7px;
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+}
+
+.header
+  .container
+  .container-left
+  .nav
+  .menus
+  .subMenu
+  .list
+  .menu:nth-child(1) {
+  font-weight: 900;
+}
+
+.header .container .container-left .nav .menus .subMenu .list .menu:hover {
+  color: #0366d6;
+}
```
이 코드들을 보면 `.header .container .container-left .nav` 속성들이 중복되어서 사용되는데 꼭 이 영역이 아니더라도 전반적으로 모든 속성들이 중복되어 사용되고 있습니다. 이런 경우 코드 가독성이 떨어져서 추후에 유지보수를 어렵게 만드는 원인이 됩니다. 작성하신 html 구조를 다시 한번 살펴보시면서 공통적으로 적용할 수 있는 속성을 다시 생각해보세요. 

또 menu와 관련된 태그가 너무 알아보기 어렵게 작성되어 있습니다. 아래 코드의 경우 `subMenu`, `list`, 'menu` 등의 속성으로 사용되고 있는데, 보다 분명한 속성명을 주셔서 가독성을 높이고, 이 속성이 표현하고자 하는 것이 무엇인지 명료하게 표기하는 것이 좋습니다. 예를 들어 같은 subMenu여도 "team-subMenu", "personal-submenu"로 표현하는 것처럼요. 
```
   <div class="subMenu">
                <ul class="list">
                  <li class="menu">
```

heejinlee07 reviewed (commented):
안녕하세요 반갑습니다 😃
첫 리뷰를 작성하시느라 고생많으셨습니다.

그리고, 현업에서는 보통 Conversation 란에 개발한 내용에 대해 기록하여 pr을 올리기 때문에
Conversation 란에 자신이 작업한 내용을 상세히 작성해주시면 좋습니다:)

ahngh0113 committed:
asset: add favicon.ico

ahngh0113 committed:
feat: add meta tag

favicon setting

ahngh0113 committed:
feat: change signin signup tag

div -> button

ahngh0113 committed:
style: apply css

button tag
class btn
class btn signup

ahngh0113 committed:
style: apply css coding convention

1. display
2. position
3. width/height
4. paddin/margin
5. border/background
6. color/font
7. etc

ahngh0113 committed:
feat: edit favicon.ioc address
